### PR TITLE
[docs] How to build compiler

### PR DESCRIPTION
**Pull request by seanshpark**
This will fill initial version of how to build compiler document

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

File: docs/howto/how-to-build-compiler.md [hunk 1]
```diff
@@ -1,7 +1,114 @@
...
+$ cmake /usr/src/gtest
+$ make
+$ sudo mv *.a /usr/lib
+
+$ pip install yapf==0.22.0 numpy
+```
+
 ## Build for Ubuntu
 
+In a typical linux development environment, including Ubuntu, you can build the runtime with a
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:64]:
```suggestion
In a typical linux development environment, including Ubuntu, you can build the compiler with a
```

File: docs/howto/how-to-build-compiler.md [hunk 2]
```diff
@@ -1,7 +1,114 @@
...
+```
+Build artifacts will be placed in `build` folder.
+
+To run unit tests:
+```
+$ ./nncc test
+```
+
+Above steps will build all the modules in the compiler folder. There are modules that are currently
+not active. To buld only as of now active modules of the compiler, we provide a preset of modules
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:81]:
```suggestion
not active. To build only as of now active modules of the compiler, we provide a preset of modules
```

File: docs/howto/how-to-build-compiler.md [hunk 3]
```diff
@@ -1,7 +1,114 @@
...
+NNCC_WORKSPACE=build/debug ./nncc configure
+./nncc build
+```
+will build debug version in `build/debug` folder, and
+
+```
+NNCC_WORKSPACE=build/release ./nncc configure -DCMAKE_BUILD_TYPE=Release
+./nncc build
+```
+will build release version in `build/releae` folder.
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:105]:
```suggestion
will build release version in `build/release` folder.
```

File: docs/howto/how-to-build-compiler.md [hunk 4]
```diff
@@ -1,7 +1,114 @@
...
+With this command, `~/.local` folder will contain all files in release.
+If you have added `~/.local/bin` in PATH, then you will now have latest compiler binaries.
+
+### Build for debug and release separately
+
+Build target folder can be customized by `NNCC_WORKSPACE` environment, as we may want to separate
+debug and release builds.
+
+```
+NNCC_WORKSPACE=build/debug ./nncc configure
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:96]:
```suggestion
$ NNCC_WORKSPACE=build/debug ./nncc configure
```

File: docs/howto/how-to-build-compiler.md [hunk 5]
```diff
@@ -1,7 +1,114 @@
...
+./nncc build
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:97]:
```suggestion
$ ./nncc build
```

File: docs/howto/how-to-build-compiler.md [hunk 6]
```diff
@@ -1,7 +1,114 @@
...
+```
+will build debug version in `build/debug` folder, and
+
+```
+NNCC_WORKSPACE=build/release ./nncc configure -DCMAKE_BUILD_TYPE=Release
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:102]:
```suggestion
$ NNCC_WORKSPACE=build/release ./nncc configure -DCMAKE_BUILD_TYPE=Release
```

File: docs/howto/how-to-build-compiler.md [hunk 7]
```diff
@@ -1,7 +1,114 @@
...
+./nncc build
```
Comment (llFreetimell) [inline docs/howto/how-to-build-compiler.md:103]:
```suggestion
$ ./nncc build
```

File: docs/howto/how-to-build-compiler.md (hunk 4 above)
Comment (seanshpark) [reply docs/howto/how-to-build-compiler.md:96]: These without `$` are to make easy copy-paste commands. Those with `$` are just examples to show outputs too.
But seems not that useful....

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by lemmaa — APPROVED** [review]
LGTM

**Pull request merged**